examples/platform/esp32/route_hook/esp_route_table.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-FileCopyrightText: 2015-2021 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "esp_err.h"
+#include "lwip/ip6_addr.h"
+#include "lwip/netif.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief OpenThread route table entry
+ *
+ */
+typedef struct
+{
+    ip6_addr_t prefix;
+    uint8_t prefix_length;
+    ip6_addr_t gateway;
+    int8_t preference;
+    uint32_t lifetime;
+    struct netif * netif;
+} esp_route_entry_t;
+
+/**
+ * @brief Adds an entry to the route table
+ *
+ * @param[in] route_entry    The route entry to be added
+ *
+ * @return
+ *   - The pointer to the added route entry on success
+ *   - NULL on failure
+ *
+ */
+esp_route_entry_t * esp_route_table_add_route_entry(const esp_route_entry_t * route_entry);
+
+/**
+ * @brief Removes an entry from the route table
+ *
+ * @param[in] route_entry    The route entry to be removed
+ *
+ * @return
+ *   - ESP_OK
+ *   - ESP_ERR_INVALID_ARG  The provided route_entry is not in the route table.
+ *
+ */
+esp_err_t esp_route_table_remove_route_entry(esp_route_entry_t * route_entry);
+
+/**
+ * @brief The lwIP ip6 route hook
```

Comment:
Could add details about when this is called, such as on a ND protocol RA being received with RIO optiona